src/main/scala/org/broadinstitute/dsde/agora/server/ga4gh/Ga4ghService.scala

```diff
@@ -3,36 +3,79 @@ package org.broadinstitute.dsde.agora.server.ga4gh
-
-            val message = QueryPublicSingle(requestContext, entity, dType)
-            perRequest(requestContext, queryHandler, message)
+    pathPrefix("ga4gh" / "v1") {
+      get {
+        path("metadata") {
+          // because of the dashes in these names, using the automatic spray json marshalling is annoying.
+          // this is a constant response anyway, so hardcode it as a string.
+          val metadataResponse:String =
+            """
+              |{
+              |  "version": "1.0.0",
+              |  "api-version": "1.0.0",
+              |  "country": "USA",
+              |  "friendly-name": "FireCloud"
+              |}
+            """.stripMargin
+          respondWithMediaType(MediaTypes.`application/json`) {
+            complete(StatusCodes.OK, metadataResponse)
           }
+        } ~
+        path("tool-classes") {
+          val toolClassesResponse:Seq[ToolClass] = Seq(ToolClass.fromEntityType(Some(AgoraEntityType.Workflow)))
+          complete(toolClassesResponse)
+        } ~
+        path("tools") { requestContext =>
+          // TODO: query params and response headers
+          val message = QueryPublicTools(requestContext)
+          perRequest(requestContext, queryHandler, message)
+        } ~
+        path("tools" / Segment) { id => requestContext =>
+          val entity = entityFromArguments(id)
+          val message = QueryPublicTool(requestContext, entity)
+          perRequest(requestContext, queryHandler, message)
+        } ~
+        path("tools" / Segment / "versions") { id => requestContext =>
+          val entity = entityFromArguments(id)
+          val message = QueryPublic(requestContext, entity)
+          perRequest(requestContext, queryHandler, message)
+        } ~
+        path("tools" / Segment / "versions" / Segment) { (id, versionId) => requestContext =>
+          val entity = entityFromArguments(id, versionId)
+          val message = QueryPublicSingle(requestContext, entity)
+          perRequest(requestContext, queryHandler, message)
+        } ~
+        path("tools" / Segment / "versions" / Segment / "dockerfile") { (id, versionId) =>
+          complete(spray.http.StatusCodes.NotImplemented)
+        } ~
+        path("tools" / Segment / "versions" / Segment / Segment / "descriptor") { (id, versionId, descriptorType) => requestContext =>
+          val entity = entityFromArguments(id, versionId)
+          val message = QueryPublicSinglePayload(requestContext, entity, parseDescriptorType(descriptorType))
+          perRequest(requestContext, queryHandler, message)
+        } ~
+        path("tools" / Segment / "versions" / Segment / Segment / "descriptor" / Segment) { (id, versionId, descriptorType, relativePath) =>
+          complete(spray.http.StatusCodes.NotImplemented)
+        } ~
+        path("tools" / Segment / "versions" / Segment / Segment / "tests") { (id, versionId, descriptorType) =>
+          complete(spray.http.StatusCodes.NotImplemented)
```

Comment:
I realize these routes could have been written more hierarchically, e.g. 
```
pathPrefix("tools") {
  pathEnd {} ~
  pathPrefix(Segment) {
    pathEnd {} ~
    pathPrefix("versions") {
      ...
    }
  }
}
```
but after going down that route, I rewrote it "flat" as you can see above; I thought it was more readable.

Reply:
I don't think we are able to implement the unimplemented endpoints - Agora just doesn't support those things. The relative-path descriptor and tests endpoints are definite no-gos. We could *possibly* implement the dockerfile endpoint, but that would require parsing the WDL and pulling out the info ... but even then, within a Workflow, each Task might have a different dockerfile, and so which should we report?

So, my plan was:
* remove swagger defs for these endpoints, so we only show swagger for what we actually support
* remove models, tests, TODOs, and any code that references these endpoints
* BUT, leave the route in place and respond with NotImplemented (instead of NotFound) just in case a power user attempts the route - this is a signal that we know the route should exist per the spec, but we don't have anything there.